File: backend/VirtualArtGallery.Core/Entities/Avatar.cs
// VirtualArtGallery.Core/Entities/Avatar.cs
using System;

namespace VirtualArtGallery.Core.Entities;

/// <summary>
/// Per-user 3D gallery avatar customization. One avatar per user (unique UserId).
/// Style fields are stored as strings for forward-compatible customization options.
/// </summary>
public class Avatar
{
    public Guid Id { get; set; }
    public string UserId { get; set; } = string.Empty;

    // Physical appearance
    public string SkinColor { get; set; } = "#E8B89E";
    public float Height { get; set; } = 1.0f;

    // Hair
    public string HairStyle { get; set; } = "short";
    public string HairColor { get; set; } = "#3B2A1F";

    // Clothing
    public string ShirtStyle { get; set; } = "tshirt";
    public string ShirtColor { get; set; } = "#3F6FB5";
    public string PantsStyle { get; set; } = "pants";
    public string PantsColor { get; set; } = "#2F2F35";
    public string ShoesColor { get; set; } = "#101015";

    // Accessories
    public string Accessory { get; set; } = "none";
    public string AccessoryColor { get; set; } = "#222222";

    // Timestamps
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public ApplicationUser User { get; set; } = null!;
}

File: web-frontend/src/types/avatar.ts
// src/types/avatar.ts
export type HairStyle = 'bald' | 'short' | 'long' | 'curly' | 'ponytail'
export type ShirtStyle = 'tshirt' | 'hoodie' | 'jacket' | 'tank'
export type PantsStyle = 'pants' | 'shorts' | 'skirt'
export type AccessoryStyle = 'none' | 'glasses' | 'sunglasses' | 'hat' | 'beanie' | 'headphones' | 'earrings' | 'mask'

export interface Avatar {
  id: string
  userId: string
  skinColor: string
  height: number
  hairStyle: HairStyle
  hairColor: string
  shirtStyle: ShirtStyle
  shirtColor: string
  pantsStyle: PantsStyle
  pantsColor: string
  shoesColor: string
  accessory: AccessoryStyle
  accessoryColor: string
  updatedAt: string
}

export interface UpdateAvatarRequest {
  skinColor?: string
  height?: number
  hairStyle?: HairStyle
  hairColor?: string
  shirtStyle?: ShirtStyle
  shirtColor?: string
  pantsStyle?: PantsStyle
  pantsColor?: string
  shoesColor?: string
  accessory?: AccessoryStyle
  accessoryColor?: string
}

export const DEFAULT_AVATAR: Avatar = {
  id: '',
  userId: '',
  skinColor: '#E8B89E',
  height: 1.0,
  hairStyle: 'short',
  hairColor: '#3B2A1F',
  shirtStyle: 'tshirt',
  shirtColor: '#3F6FB5',
  pantsStyle: 'pants',
  pantsColor: '#2F2F35',
  shoesColor: '#101015',
  accessory: 'none',
  accessoryColor: '#222222',
  updatedAt: new Date().toISOString(),
}